JobRecommendationService.md

```diff
@@ -0,0 +1,173 @@
+# Job Recommendation service
+# Contents
+
+- [Summary](#summary)
+- [Request structure](#request-structure)
+  - [JobData](#jobdata)
+    - [Job](#job)
+  - [UserData](#userdata)
+  - [UserPreferences](#userpreferences)
+    - [PreferredLocation](#preferredlocation)
+    - [Compensation](#compensation)
+    - [CaroteneTitle](#carotenetitle)
+    - [CaroteneVersion](#caroteneversion)
+- [Response Structure](#response-structure)
+- [Versioning](#versioning)
+
+----------------------------
+# Summary
+
+Job Recommendation Service is part of the profile recommendations, which recommends similar and related jobs to a user based on the jobs he applied and user preferences. The recommendations are combined from FAISS indexes and GBR data, then filtered by the applied jobs.
+
+The endpoint for Job Recommendation Service is at https://api.careerbuilder.com/core/recommendations/job
+
+----------------------------
+# Request structure
+The service supports GET and POST requests to:
+https://api.careerbuilder.com/core/search/candidatereranker.
+
+OAUTH credentials are **required**.
+
+A request is composed of 3 main parts:
+`job_data`, `user_data`, `user_preferences`.
+
+| Param    | Type | Required | Description |
+|----------|------|----------|-------------|
+| `job_data` | [**JobData**](#jobdata) | **TRUE** | **JobData** contains a list of jobs which is needed for recommendations.
+| `user_data` | [**UserData**](#userdata) | **FALSE** | **UserData** contains visitor id and email address of the user, which is then used to retrieve applied jobs for filtering.
+| `user_preferences` | [**UserPreferences**](#userpreferences) | **FALSE** | **UserPreferences** indicates preferences of the user including location, compensation, job titles in carotenes and the associated varotene version.
+
+### JobData
+| Param    | Type | Required | Description |
+|----------|------|----------|-------------|
+| `source_jobs` | List<[**Job**](#job)> | **TRUE** | A list of **Job**.
+
+Note: currently only the first job in the list is used for recommendations.
+
+#### Job
+| Param    | Type | Required | Description |
+|----------|------|----------|-------------|
+| `job_did` | String | **TRUE** | This String is required to start with `J` and have a length that is less than the given `MAX_LENGTH`, specifically 22 currently.
+
+### UserData
+| Param    | Type | Required | Description |
+|----------|------|----------|-------------|
+| `visitor_id` | String | **FALSE** | This starts with `VX` and is required to be shorter than 50 (`MAX_LENGTH`) characters .
+| `email_address` | String | **TRUE** | The email of the registered user.
+
+**UserData** contains two String parameters: `visitor_id` and `email_address`. However, currently only `email_address` is used to retrieve user applications.
```

Comment:
I think it might be a little better to put these summaries before their corresponding table rather than after them.